### DISCUSSION TITLE
How to add RT-thread msh user command?

### DISCUSSION THREAD

**Discussion by danilliu-sudo** [onelife/Arduino_RT-Thread#10] [Q&A] [answered]
hi, onelife.

I'm trying the Arduino_RT_Thread on my DUE board, It works good. 
but when I using MSH_CMD_EXPORT() command  to add my owe msh command,  the Compile and download is ok, but msh does not recognize my command. (help has no display, and running the command will display "command not fount".
how can I work out this problem? 
thanks. 

my code is such as：

void disp(void) {
。。。
}
MSH_CMD_EXPORT(disp, display sample data);

**Comment by onelife** [comment, marked as answer]
Hi @danilliu-sudo,

There is an example of user commands, https://github.com/onelife/Arduino_RT-Thread/blob/v0.9.4/examples/FinSH/FinSH.ino:
https://github.com/onelife/Arduino_RT-Thread/blob/2906ad832b44149f14c707018f349a5b00cfa7d7/examples/FinSH/FinSH.ino#L68-L95

Uncomment the following line:
https://github.com/onelife/Arduino_RT-Thread/blob/2906ad832b44149f14c707018f349a5b00cfa7d7/src/shell_cmd.h#L138

**Notes on using MSH user commands**
- Enable flag `CONFIG_USING_FINSH` and `CONFIG_USING_MSH`.
- The command has to be defined as a C function.
   In `.ino` file, the function has to be wrapped in `extern "C" { ... }`.
- The function prototype is `int the_command(int argc, char **argv)`.
- Add a line, `ADD_MSH_CMD( ... )`, in https://github.com/onelife/Arduino_RT-Thread/blob/v0.9.4/src/shell_cmd.h.
